tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java

```diff
@@ -95,6 +95,23 @@ public void testObjectMocking() throws BallerinaTestException {
         }
     }
 
+    /**
+     * Test object mock using test double when client implementation contains a resource function.
+     *
+     * @throws BallerinaTestException
+     */
+    @Test()
+    public void testObjectMockDouble() throws BallerinaTestException {
+        String msg1 = "1 passing";
+        String[] args = mergeCoverageArgs(new String[]{"object-mocking-tests2"});
+        String output = balClient.runMainAndReadStdOut("test", args,
+                new HashMap<>(), projectPath, false);
+        if (!output.contains(msg1)) {
+            Assert.fail("Test failed due to object mocking failure with resource functions in test framework." +
```

Comment:
Can we store the complete output in a file and carry out the test against it??

Reply:
Decided to address this with https://github.com/ballerina-platform/ballerina-lang/pull/39562

Reply:
We will address this comment in https://github.com/ballerina-platform/ballerina-lang/issues/39561.